src/plugins/home/public/application/components/home_app.js

```diff
@@ -78,7 +69,7 @@ export function HomeApp({ directories, solutions }) {
               hasUserIndexPattern={() => indexPatternService.hasUserIndexPattern()}
             />
           </Route>
-          <Route path="*" exact={true} component={RedirectToDefaultApp} />
+          <Redirect to="/" />
```

Comment:
:information_source: This adds a redirect to the home page in case any unknown route has been entered. With the old logic the existing `Route` would have taken care of forwarding to the `defaultAppId` in this case.

Reply:
Why not redirect to `defaultRoute` set in uiSettings?

Reply:
The question is: why would only the Home page do that in this case, and not ALL apps? I feel it's a bit weird, that something like `/app/home#/fooobar` would bring you to the default app, but `/app/dashboard#/foobar` wouldn't. Or we'd basically need to set this as an expactation to all apps, that if they can't find the route in their app they should forward to the default app, which sounds like a larger change to me. I am happy to rediscuss this behavior, but I think we should not make a special case JUST for the home app and nothing else, and in this case rather let every app handle their "unknown route" internally to do something reasonable.

Reply:
> Or we'd basically need to set this as an expactation to all apps, that if they can't find the route in their app they should forward to the default app, which sounds like a larger change to me.

agree. that sounds like a high-level policy.